File: Client/Render/RenderWeepingAngel.java
package WeepingAngels.Client.Render;

import WeepingAngels.Client.Model.ModelWeepingAngel;
import WeepingAngels.Entity.EntityWeepingAngel;
import WeepingAngels.lib.Reference;
import net.minecraft.client.renderer.entity.RenderLiving;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityLiving;
import net.minecraft.util.ResourceLocation;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class RenderWeepingAngel extends RenderLiving
{
	protected ModelWeepingAngel weepingAngelModel;
	
    public RenderWeepingAngel(float f)
    {
        super(new ModelWeepingAngel(), f);
        this.weepingAngelModel = (ModelWeepingAngel)this.mainModel;
    }
    
    public void renderWeepingAngel(EntityWeepingAngel entityweepingangel, double d, double d1, double d2, 
            float f, float f1)
    {
        super.doRenderLiving(entityweepingangel, d, d1, d2, f, f1);
    }

    public void doRenderLiving(EntityLiving entityliving, double d, double d1, double d2, 
            float f, float f1)
    {
        renderWeepingAngel((EntityWeepingAngel)entityliving, d, d1, d2, f, f1);
    }

    public void doRender(Entity entity, double d, double d1, double d2, 
            float f, float f1)
    {
        renderWeepingAngel((EntityWeepingAngel)entity, d, d1, d2, f, f1);
    }

	@Override
	protected ResourceLocation getEntityTexture(Entity entity) {
		return Reference.weepingAngelTex;
	}

	
	
}

File: src/main/scala/com/temportalist/weepingangels/client/render/RenderWeepingAngel.scala
package com.temportalist.weepingangels.client.render

import com.temportalist.weepingangels.client.render.models.ModelWeepingAngel
import com.temportalist.weepingangels.common.WAOptions
import net.minecraft.client.renderer.entity.{RenderLiving, RenderManager}
import net.minecraft.entity.{Entity, EntityLiving}
import net.minecraft.util.ResourceLocation
import net.minecraftforge.fml.relauncher.{Side, SideOnly}

/**
 *
 *
 * @author TheTemportalist
 */
@SideOnly(Side.CLIENT)
class RenderWeepingAngel(manager: RenderManager)
		extends RenderLiving(manager, new ModelWeepingAngel(), 0.5F) {

	var texture: ResourceLocation = WAOptions.weepingAngel1

	override def getEntityTexture(entity: Entity): ResourceLocation = {
		//println ("get texture")
		this.texture
	}

	override def doRender(entity: Entity, x: Double, y: Double, z: Double, p_76986_8_ : Float,
			partialTicks: Float): Unit = {
		//println ("render e")
		super.doRender(entity, x, y, z, p_76986_8_, partialTicks)
	}

	override def doRender(entity: EntityLiving, d1: Double, d2: Double, d3: Double, f1: Float,
			f2: Float): Unit = {


		//println()
		this.bindEntityTexture(entity)

		super.doRender(entity, d1, d2, d3, f1, f2)
	}

	/*
	override def bindEntityTexture(entity: Entity): Boolean = {
		entity match {
			case angel: EntityWeepingAngel =>
				GlStateManager.bindTexture(angel.getTextureID())
				true
			case _ =>
				false
		}
	}
	*/

}
